src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/config/Config.java

```diff
@@ -413,6 +471,44 @@ public void setValues(JSONObject formData) {
         replicationConfig = ReplicationConfig.createReplicationConfigFromJSON(formData);
     }
 
+    private void initializeGerritCommands() {
+        List<String> stringLabels = new ArrayList<String>(Arrays.asList("verified", "code-review"));
+
+        for (VerdictCategory label : categories) {
+            stringLabels.add(label.getVerdictValue().toLowerCase());
+
+            if (!label.getVerdictValue().equalsIgnoreCase(CODE_REVIEW) && !label.getVerdictValue().equalsIgnoreCase(VERIFIED)) {
+                String commandAddition = " --" + label.getVerdictValue().toLowerCase() + " <" + label.getVerdictValue().toUpperCase() + "> ";
+                gerritVerifiedCmdBuildStarted = gerritVerifiedCmdBuildStarted.contains(commandAddition) ? gerritVerifiedCmdBuildStarted :
+                    (gerritVerifiedCmdBuildStarted += commandAddition);
+
+                gerritVerifiedCmdBuildSuccessful = gerritVerifiedCmdBuildSuccessful.contains(commandAddition) ? gerritVerifiedCmdBuildSuccessful :
+                    (gerritVerifiedCmdBuildSuccessful += commandAddition);
+
+                gerritVerifiedCmdBuildFailed = gerritVerifiedCmdBuildFailed.contains(commandAddition) ? gerritVerifiedCmdBuildFailed :
+                    (gerritVerifiedCmdBuildFailed += commandAddition);
+
+                gerritVerifiedCmdBuildUnstable = gerritVerifiedCmdBuildUnstable.contains(commandAddition) ? gerritVerifiedCmdBuildUnstable :
+                    (gerritVerifiedCmdBuildUnstable += commandAddition);
+
+                gerritVerifiedCmdBuildNotBuilt = gerritVerifiedCmdBuildNotBuilt.contains(commandAddition) ? gerritVerifiedCmdBuildNotBuilt :
+                    (gerritVerifiedCmdBuildNotBuilt += commandAddition);
+            }
+        }
```

Comment:
It is probably simpler, or a bit more straight forward at least, if all the templated `--code-review <CODE-REVIEW> ...` was replaced with just a <VERDICT-PARAMS> or similar that the `ParameterExpander` would later expand to all those parameters.
The `readResolve` can do the replacement in any existing configured commands.
wdyt?

Reply:
I remember though that there are some searching going on in the rest commands to extract the data out of the commands that might need some changes as well if you haven't already.

Reply:
At that time, this was the "least invasive" way to add the commands for additional labels. But maybe it would be an improvement for a future commit?

Reply:
Well this change is quite invasive as it is, so slightly more invasive wouldn't do much difference I think :)